### DISCUSSION TITLE
JSON Schema Field Description

### DISCUSSION THREAD

**Discussion by Bodekaer** [RageAgainstThePixel/com.openai.unity#288] [Q&A]
Hi @StephenHodgson, 
Can you clarify how to include a field description with the new strict/structured format? 

Example I tried, but doesn't send the description to the API:
```
[JsonProperty("name")]
[Description("Name of the user")]
public string Name { get; set; }
```

**Comment by StephenHodgson** [comment]
There's currently no way to add a description.

**Reply by StephenHodgson** [reply]
You could try reusing the description from the Function attributes in the meantime. But I didn't consider this too important.

**Comment by Bodekaer** [comment]
I'm getting a different odd error now. Have you seen this where the `arguments` field is set only for some tools? In this case only the 2nd tool gets it defined. Not sure why it happens.

Request:
```
    "body": {
      "model": "gpt-4o",
      "description": "...",
      "instructions": "...",
      "tools": [
        {
          "type": "file_search"
        },
        {
          "type": "function",
          "function": {
            "name": "WorksFine",
            "description": "...",
            "parameters": {
              "type": "object",
              "properties": {
                "maxDistanceInMeters": {
                  "type": "number"
                }
              },
              "required": [
                "maxDistanceInMeters"
              ],
              "additionalProperties": false
            },
            "strict": true
          }
        },
        {
          "type": "function",
          "function": {
            "name": "DoesNotWork",
            "description": "...",
            "parameters": {
              "type": "object",
              "properties": {
                "somePropertyName": {
                  "type": "string"
                }
              },
              "required": [
                "somePropertyName"
              ],
              "additionalProperties": false
            },
            "arguments": "{\"somePropertyName\":\"property value here\"}",
            "strict": true
          }
        }
```

Error message from API:
```
      "error": {
        "message": "Unknown parameter: 'tools[4].function.arguments'.",
        "type": "invalid_request_error",
        "param": "tools[4].function.arguments",
        "code": "unknown_parameter"
      }
```

**Reply by StephenHodgson** [reply]
Haven't seen it but if you can get me repro steps I'll look into it

**Reply by Bodekaer** [reply]
Update: Turns out it was caused when a Tool/Function was called async, before the Assistant had been fully initialized.

When the tool is called, the `Function.Arguments` field is set, and that was then being included in the CreateAssistant request. 
More specifically, it happened because I have "Domain Refresh" turned off for faster development/compile time, and in this case, it meant if I had call the tool in the previous run/session, and exited play mode, then the `Function.Argument` value was still set.

I'm properly clearing the tool cache manually now each time Unity exits play mode, which fixes it.

**Reply by Bodekaer** [reply]
And I realize now, with setting specific Run.Tools, the Tool.Function is passed in with the `Arguments` field as well, causing the same bug.
E.g. to replicate:
```
CreateAssistant
CreateRun (with Tool defined)
CreateRun (with same Tool defined) (this time the Arguments field value from the first CreateRun is included)
```

Can I put some tag on the `Function.Arguments` field, so it doesn't get included in the JSON Schema?

**Reply by StephenHodgson** [reply]
I don't think the Arguments should be included the second time. Let's file this as a proper bug.

**Reply by StephenHodgson** [reply]
@Bodekaer https://github.com/RageAgainstThePixel/OpenAI-DotNet/issues/360
